example/tutorial_quanconv_cifar10.py

```diff
@@ -0,0 +1,140 @@
+import numpy as np
+import tensorflow as tf
+import tensorlayer as tl
+
+bitW = 8
+bitA = 8
+
+tf.logging.set_verbosity(tf.logging.DEBUG)
+tl.logging.set_verbosity(tl.logging.DEBUG)
+
+sess = tf.InteractiveSession()
+
+X_train, y_train, X_test, y_test = tl.files.load_cifar10_dataset(shape=(-1, 32, 32, 3), plotable=False)
+
+
+def model(x, y_, reuse, is_train, bitW, bitA):
+    with tf.variable_scope("model", reuse=reuse):
+        net = tl.layers.InputLayer(x, name='input')
+        net = tl.layers.QuanConv2dWithBN(
+            net, 64, (5, 5), (1, 1), act=tf.nn.relu, padding='SAME', is_train=is_train, bitW=bitW, bitA=bitA,
+            name='qcnnbn1'
+        )
+        net = tl.layers.MaxPool2d(net, (3, 3), (2, 2), padding='SAME', name='pool1')
+        # net = tl.layers.BatchNormLayer(net, act=tl.act.htanh, is_train=is_train, name='bn1')
+        net = tl.layers.QuanConv2dWithBN(
+            net, 64, (5, 5), (1, 1), padding='SAME', act=tf.nn.relu, is_train=is_train, bitW=bitW, bitA=bitA,
+            name='qcnnbn2'
+        )
+        # net = tl.layers.BatchNormLayer(net, act=tl.act.htanh, is_train=is_train, name='bn2')
+        net = tl.layers.MaxPool2d(net, (3, 3), (2, 2), padding='SAME', name='pool2')
+        net = tl.layers.FlattenLayer(net, name='flatten')
+        net = tl.layers.QuanDenseLayer(net, 384, act=tf.nn.relu, bitW=bitW, bitA=bitA, name='qd1relu')
+        net = tl.layers.QuanDenseLayer(net, 192, act=tf.nn.relu, bitW=bitW, bitA=bitA, name='qd2relu')
+        net = tl.layers.DenseLayer(net, 10, act=None, name='output')
+        y = net.outputs
+
+        ce = tl.cost.cross_entropy(y, y_, name='cost')
+        L2 = 0
+        for p in tl.layers.get_variables_with_name('relu/W', True, True):
+            L2 += tf.contrib.layers.l2_regularizer(0.004)(p)
+        cost = ce + L2
+
+        # correct_prediction = tf.equal(tf.argmax(tf.nn.softmax(y), 1), y_)
+        correct_prediction = tf.equal(tf.cast(tf.argmax(y, 1), tf.int64), y_)
+        acc = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+
+        return net, cost, acc
+
+
+def distort_fn(x, is_train=False):
+    x = tl.prepro.crop(x, 24, 24, is_random=is_train)
+    if is_train:
+        x = tl.prepro.flip_axis(x, axis=1, is_random=True)
+        x = tl.prepro.brightness(x, gamma=0.1, gain=1, is_random=True)
+    x = (x - np.mean(x)) / max(np.std(x), 1e-5)  # avoid values divided by 0
+    return x
+
+
+x = tf.placeholder(dtype=tf.float32, shape=[None, 24, 24, 3], name='x')
+y_ = tf.placeholder(dtype=tf.int64, shape=[None], name='y_')
+
+network, cost, _ = model(x, y_, False, True, bitW=bitW, bitA=bitA)
+_, cost_test, acc = model(x, y_, True, False, bitW=bitW, bitA=bitA)
+
+# train
+n_epoch = 50000
+learning_rate = 0.0001
+print_freq = 1
+batch_size = 128
+
+train_op = tf.train.AdamOptimizer(learning_rate, beta1=0.9, beta2=0.999, epsilon=1e-08,
+                                  use_locking=False).minimize(cost)
+
+tl.layers.initialize_global_variables(sess)
```

Comment:
This function will be deprecated soon, see
https://github.com/tensorlayer/tensorlayer/blob/master/tensorlayer/layers/utils.py#L198
for alternative solution.

Reply:
+1